x-pack/plugins/cases/server/services/attachments/index.test.ts

```diff
@@ -537,4 +537,39 @@ describe('AttachmentService', () => {
       });
     });
   });
+
+  describe('countPersistableStateAndExternalReferenceAttachments', () => {
+    it('does not throw and calls unsecuredSavedObjectsClient.find with the right parameters', async () => {
+      unsecuredSavedObjectsClient.find.mockResolvedValue(
+        createSOFindResponse([{ ...createUserAttachment(), score: 0 }])
+      );
+
+      await expect(
+        service.countPersistableStateAndExternalReferenceAttachments({ caseId: 'test-id' })
+      ).resolves.not.toThrow();
+
+      await expect(unsecuredSavedObjectsClient.find).toHaveBeenCalledWith(
+        expect.objectContaining({
```

Comment:
I think we should validate the filter passed to `unsecuredSavedObjectsClient.find`. You can do 

```
expect(unsecuredSavedObjectsClient.find.mock.calls[0][0]).toMatchInlineSnapshot(...)
```